waspc/data/Generator/templates/react-app/src/auth/pages/DefaultPageLoader.jsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+// TODO: Should I take this out of auth/, into more general place?
+export const DefaultPageLoader = ({status, error}) => {
```

Comment:
I would love to throw some TS here, to specify status is supposed to be `'loading' | 'error'`. How should I go about that though? Provide them with a type they can use? Provide them with a hook they use to define this component? Any advice, something I should look at and copy?

Reply:
> Provide them with a type they can use? 

You simply type the props of the function to be:
```ts
export function DefaultPageLoader(
  { status, error }: { status: 'loading' | 'error';  error: { message?: string } }
) { ... }
```

Reply:
Oh, you want to provide them a way to define a component that will fit in our place of use i.e. [createAuthRequiredPage.jsx](https://github.com/wasp-lang/wasp/pull/1968/files#diff-8cb8bc1efac0cbcc354c11abe7f36616f7096a8369e6282eecf51cb1077f08ee)

Using some function like `definePageLoaderComponent` could be a could idea!

Reply:
Yup that is what I meant! But ok, not doing this now, still thanks for letting me know.